File: app/src/test/java/no/nav/tilbakemeldingsmottak/ReadXslxFile.java
package no.nav.tilbakemeldingsmottak;


import org.apache.poi.ss.usermodel.Row;
import org.apache.poi.ss.usermodel.Sheet;
import org.apache.poi.ss.usermodel.Workbook;
import org.apache.poi.ss.usermodel.WorkbookFactory;

import java.io.File;
import java.io.FileInputStream;
import java.io.IOException;
import java.util.ArrayList;
import java.util.HashMap;
import java.util.List;
import java.util.Map;

public class ReadXslxFile {
    public Map<Integer, List<String>> readExcelFile(String filePath) throws IOException {
        FileInputStream inputStream = null;
        try {
            Map<Integer, List<String>> resultMap = new HashMap<>();
            ClassLoader classLoader = getClass().getClassLoader();
            File file = new File(classLoader.getResource(filePath).getFile());

            inputStream = new FileInputStream(file);
            Workbook workbook = WorkbookFactory.create(inputStream);
            Sheet sheet = workbook.getSheetAt(0);

            int noOfColumns = sheet.getRow(0).getPhysicalNumberOfCells();
            for (Row row : sheet) {
                int rowNum = row.getRowNum();
                List<String> cellList = new ArrayList<>();

                for (int i = 0; i < noOfColumns; i++) {
                    cellList.add(row.getCell(i) == null ? "" : row.getCell(i).toString());
                }

                resultMap.put(rowNum, cellList);
            }

            workbook.close();
            inputStream.close();

            return resultMap;
        } finally {
            if (inputStream != null) inputStream.close();
        }
    }
}

File: app/src/test/kotlin/no/nav/tilbakemeldingsmottak/util/ReadXslxFile.kt
package no.nav.tilbakemeldingsmottak.util

import org.apache.poi.ss.usermodel.WorkbookFactory
import java.io.File
import java.io.FileInputStream

internal class ReadXslxFile {
    fun readExcelFile(filePath: String): Map<Int, List<String>> {
        var inputStream: FileInputStream? = null
        return try {
            val resultMap = mutableMapOf<Int, List<String>>()
            val classLoader = javaClass.classLoader
            val file =
                File(classLoader.getResource(filePath)?.file ?: throw IllegalArgumentException("Resource not found"))
            inputStream = FileInputStream(file)
            val workbook = WorkbookFactory.create(inputStream)
            val sheet = workbook.getSheetAt(0)
            val noOfColumns = sheet.getRow(0).physicalNumberOfCells

            for (row in sheet) {
                val rowNum = row.rowNum
                val cellList = mutableListOf<String>()
                for (i in 0 until noOfColumns) {
                    cellList.add(row.getCell(i)?.toString() ?: "")
                }
                resultMap[rowNum] = cellList
            }
            workbook.close()
            resultMap
        } finally {
            inputStream?.close()
        }
    }
}
